### DISCUSSION TITLE
Love this, very handy app! Thank you!

### DISCUSSION THREAD

**Discussion by forrest-herman** [ther0n/UnnaturalScrollWheels#53] [General]
Out of curiosity, where did you learn to create an app like this? I would love to start creating Mac Apps as well!

**Comment by ther0n** [comment]
I don't have a lot of advice for you unfortunately. This was a weekend project which originally was only going to be for my personal use. I got a job as a full-time developer and knew I'd be using a Mac 8 hours a day. For development I want to plug into an external monitor(s), keyboard, and mouse and I was so used to how scrolling worked on Linux/Windows and found it very annoying on macOS. I also wanted to have a way to disable mouse acceleration so that my mouse moves the same amount on the screen regardless of which OS I'm using at the time. So the weekend before I started the new job, I decided to figure out a solution so I didn't have to retrain my brain every time I switched OSes

I learned everything I needed as I went. As far as figuring out how to intercept scroll wheel events (which is the only hard part really), I looked at the code for [discrete-scroll](https://github.com/emreyolcu/discrete-scroll). It was written in objective-c so I spent a good amount of time trying to figure out how to do it in Swift. Once I did that, I read the documentation for the `CGEventField.scrollWheelEvent*` fields. By 'read' I mean more that I learned the variables existed at all as the [documentation](https://developer.apple.com/documentation/coregraphics/cgeventfield/scrollwheeleventdeltaaxis1) on the fields the app modifies didn't/doesn't have any explanation as to what they do. I inferred based on their names and figured out what they did through trial and error. At this point it had no GUI and was configured using the `defaults` command (explained more [here](https://macos-defaults.com/)) but had all the functionality I needed. Then I decided to add a GUI and polish it up so that other people could use it which took another several hours. Once that was done, I released it!

**Comment by la5rocks** [comment]
This is a game-changer! The broken scroll direction preferences are so frustrating.

I wasn't even aware how much the acceleration was messing me up. Mouse movement has felt so sluggish, and now is nice and snappy like I expected it to be.

Now if we can just re-map the home and end keys, LOL

**Comment by Doogiemuc** [comment]
Love this SO much! Thanx!!!!!

**Comment by deathwinger** [comment]
This is an amazing piece of open source coding, implementation of users' needs and enhancing their experience. Something that all of Apple and even paid apps have yet to get right has been achieved right here. 

Apple, hire this team to implement your peripheral software needs please, they understand. 

All of this to say, thank you for creating this!
